core/sched.cc

```diff
@@ -289,8 +290,26 @@ void cpu::reschedule_from_interrupt(bool called_from_yield,
             return;
 #endif
         } else if (!called_from_yield) {
-            auto &t = *runqueue.begin();
-            if (p->_runtime.get_local() < t._runtime.get_local()) {
+           auto &t = *runqueue.begin();
+            if (p->_realtime._priority > 0) {
+                // Only switch to a higher-priority realtime thread, or to
+                // the next one with equal priority if yielding.
+                if (t._realtime._priority < p->_realtime._priority) {
+#ifdef __aarch64__
+                    return switch_data;
+#else
+                    return;
+#endif
+                }
+                if (t._realtime._priority == p->_realtime._priority &&
```

Comment:
Is this condition right? If the priority of `t` is equal to `p`, then we keep `p` running. Is this correct?